### PR TITLE
FINERACT-1932: Fineract modularization - fineract-tax

### PR DESCRIPTION
**Pull request by alberto-art3ch**
## Description

Create the initial fineract-tax modules - used as tax domain module classes, interfaces and services

[FINERACT-1932](https://issues.apache.org/jira/browse/FINERACT-1932)

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/jobs/service/StuckJobExecutorServiceImpl.java [hunk 1]
```diff
@@ -102,7 +102,7 @@ protected void doInTransactionWithoutResult(TransactionStatus status) {
 
     private void waitUntilAllPartitionsFinished(Long stuckJobId, String partitionerStepName) throws InterruptedException {
         while (!areAllPartitionsCompleted(stuckJobId, partitionerStepName)) {
-            log.info("Sleeping for a second to wait for the partitions to complete for job {}", stuckJobId);
+            log.info("Sleeping for a second to wait for the partitions to complete for job {} {}", stuckJobId, partitionerStepName);
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/jobs/service/StuckJobExecutorServiceImpl.java]: This seems not relevant change. Please move it into separate PR!

File: fineract-provider/src/main/resources/db/changelog/tenant/parts/0026_reversals_for_reversed_transactions.xml [hunk 2]
```diff
@@ -23,6 +23,7 @@
                    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.1.xsd">
     <changeSet author="fineract" id="1">
+        <validCheckSum>9:75966e6d941a3177d7b40612b4837cae</validCheckSum>
```
Comment (adamsaghy) [inline fineract-provider/src/main/resources/db/changelog/tenant/parts/0026_reversals_for_reversed_transactions.xml]: Why do we need new checksum?

File: fineract-provider/src/main/resources/db/changelog/tenant/parts/0071_add_external_id_support_for_loan_transaction.xml [hunk 3]
```diff
@@ -23,6 +23,7 @@
                    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.3.xsd">
     <changeSet author="fineract" id="1">
+        <validCheckSum>9:9f188ad9a36d71e678fe9f25d0dd25b3</validCheckSum>
```
Comment (adamsaghy) [inline fineract-provider/src/main/resources/db/changelog/tenant/parts/0071_add_external_id_support_for_loan_transaction.xml]: Why do we need new checksum?

File: fineract-tax/src/main/resources/jpa/taxes/persistence.xml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        <class>org.apache.fineract.portfolio.floatingrates.domain.FloatingRatePeriod</class>
+        <class>org.apache.fineract.portfolio.calendar.domain.Calendar</class>
+        <class>org.apache.fineract.portfolio.calendar.domain.CalendarHistory</class>
+        <class>org.apache.fineract.portfolio.calendar.domain.CalendarInstance</class>
+        <class>org.apache.fineract.useradministration.domain.AppUser</class>
+        <class>org.apache.fineract.useradministration.domain.Role</class>
+        <class>org.apache.fineract.useradministration.domain.Permission</class>
+        <class>org.apache.fineract.useradministration.domain.AppUserClientMapping</class>
+        <!-- Fineract Charge module -->
+        <class>org.apache.fineract.portfolio.charge.domain.Charge</class>
```
Comment (adamsaghy) [inline fineract-tax/src/main/resources/jpa/taxes/persistence.xml]: Tax module depends on charge? I see only fineract-core as dependency...

**Review by adamsaghy — CHANGES_REQUESTED** [review]
Please kindly check my review!

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/jobs/service/StuckJobExecutorServiceImpl.java (hunk 1 above)
Comment (alberto-art3ch) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/jobs/service/StuckJobExecutorServiceImpl.java]: Done! removed

File: fineract-provider/src/main/resources/db/changelog/tenant/parts/0026_reversals_for_reversed_transactions.xml (hunk 2 above)
Comment (alberto-art3ch) [reply fineract-provider/src/main/resources/db/changelog/tenant/parts/0026_reversals_for_reversed_transactions.xml]: Done! removed

File: fineract-provider/src/main/resources/db/changelog/tenant/parts/0071_add_external_id_support_for_loan_transaction.xml (hunk 3 above)
Comment (alberto-art3ch) [reply fineract-provider/src/main/resources/db/changelog/tenant/parts/0071_add_external_id_support_for_loan_transaction.xml]: Done! removed

File: fineract-tax/src/main/resources/jpa/taxes/persistence.xml (hunk 4 above)
Comment (alberto-art3ch) [reply fineract-tax/src/main/resources/jpa/taxes/persistence.xml]: Done! updated

**Comment by alberto-art3ch** [comment]
> Please kindly check my review!

Comments attended, Thanks!

File: fineract-provider/src/main/resources/db/changelog/tenant/parts/0071_add_external_id_support_for_loan_transaction.xml (hunk 3 above)
Comment (adamsaghy) [reply fineract-provider/src/main/resources/db/changelog/tenant/parts/0071_add_external_id_support_for_loan_transaction.xml]: Seems you have missed one! ;)

**Review by adamsaghy — CHANGES_REQUESTED** [review]
Please kindly check my review!

File: fineract-provider/src/main/resources/db/changelog/tenant/parts/0071_add_external_id_support_for_loan_transaction.xml (hunk 3 above)
Comment (alberto-art3ch) [reply fineract-provider/src/main/resources/db/changelog/tenant/parts/0071_add_external_id_support_for_loan_transaction.xml]: Done! removed

**Comment by adamsaghy** [comment]
@alberto-art3ch please resolve the conflicts! ;)

**Comment by alberto-art3ch** [comment]
> @alberto-art3ch please resolve the conflicts! ;)

Done! conflicts solved

**Review by adamsaghy — APPROVED** [review]
LGTM
